src/main/java/picard/illumina/IlluminaBasecallsToSam.java

```diff
@@ -282,28 +297,42 @@ private void initialize() {
         }
 
         if (OUTPUT != null) {
-            barcodeSamWriterMap.put(null, buildSamFileWriter(OUTPUT, SAMPLE_ALIAS, LIBRARY_NAME, buildSamHeaderParameters(null)));
+            barcodeSamWriterMap.put(null, buildSamFileWriter(OUTPUT, SAMPLE_ALIAS, LIBRARY_NAME, buildSamHeaderParameters(null), !USE_NEW_CONVERTER));
         } else {
             populateWritersFromLibraryParams();
         }
 
         final int numOutputRecords = readStructure.templates.length();
-
-        basecallsConverter = new IlluminaBasecallsConverter<SAMRecordsForCluster>(BASECALLS_DIR, BARCODES_DIR, LANE, readStructure,
-                barcodeSamWriterMap, true, MAX_READS_IN_RAM_PER_TILE/numOutputRecords, TMP_DIR, NUM_PROCESSORS, FORCE_GC,
-                FIRST_TILE, TILE_LIMIT, new QueryNameComparator(), new Codec(numOutputRecords), SAMRecordsForCluster.class,
-                bclQualityEvaluationStrategy, this.APPLY_EAMSS_FILTER, INCLUDE_NON_PF_READS, IGNORE_UNEXPECTED_BARCODES);
-
-        log.info("DONE_READING STRUCTURE IS " + readStructure.toString());
-
         // Combine any adapters and custom adapter pairs from the command line into an array for use in clipping
         final List<AdapterPair> adapters = new ArrayList<>();
         adapters.addAll(ADAPTERS_TO_CHECK);
         if (FIVE_PRIME_ADAPTER != null && THREE_PRIME_ADAPTER != null) {
             adapters.add(new CustomAdapterPair(FIVE_PRIME_ADAPTER, THREE_PRIME_ADAPTER));
         }
 
-        /**
+        if (USE_NEW_CONVERTER) {
+            final MetricsFile<BarcodeMetric, Integer> metrics = getMetricsFile();
+
+            if (METRICS_FILE != null) {
+                IOUtil.assertFileIsWritable(METRICS_FILE);
+            } else {
+                log.warn("No METRIC_FILE specified. Barcode metrics will not be generated.");
+            }
+            basecallsConverter = new NewIlluminaBasecallsConverter<>(BASECALLS_DIR, BARCODES_DIR, LANE, readStructure,
```

Comment:
If we don't use the type argument to the generic, does it need to be declared as a generic? It doesn't seem like it's buying us anything unless a type is used.